File: svws-webclient/core/src/asd/validate/schueler/ValidatorSses00SchuelerStammdatenErsteStaatsangehoerigkeit.ts
import type { Supplier } from '../../../java/util/function/Supplier';
import { Class } from '../../../java/lang/Class';
import { ValidatorSses01SchuelerStammdatenErsteStaatsangehoerigkeit } from '../../../asd/validate/schueler/ValidatorSses01SchuelerStammdatenErsteStaatsangehoerigkeit';
import { ValidatorKontext } from '../../../asd/validate/ValidatorKontext';
import { Validator } from '../../../asd/validate/Validator';

export class ValidatorSses00SchuelerStammdatenErsteStaatsangehoerigkeit extends Validator {

	/**
	 * Die Staatsangehoerigkeit des Schuelers
	 */
	private readonly _idStaatsangehoerigkeit: Supplier<number | null>;


	/**
	 * Erstellt einen neuen Validator mit den übergebenen Daten und dem übergebenen Kontext
	 *
	 * @param idStaatsangehoerigkeit    StaatsangehörigkeitID
	 * @param kontext                   der Kontext des Validators
	 */
	public constructor(idStaatsangehoerigkeit: Supplier<number | null>, kontext: ValidatorKontext) {
		super(kontext);
		this._idStaatsangehoerigkeit = idStaatsangehoerigkeit;
		this._validatoren.add(new ValidatorSses01SchuelerStammdatenErsteStaatsangehoerigkeit(this.getNotNullSupplierLong(idStaatsangehoerigkeit), kontext));
	}

	protected pruefe(): boolean {
		const idStaatsangehoerigkeit: number | null = this._idStaatsangehoerigkeit.get();
		if (idStaatsangehoerigkeit === null) {
			this.addFehler(0, "1. Staatsangehörigkeit des Schülers: Kein Wert vorhanden.");
			return false;
		}
		return true;
	}

	transpilerCanonicalName(): string {
		return 'de.svws_nrw.asd.validate.schueler.ValidatorSses00SchuelerStammdatenErsteStaatsangehoerigkeit';
	}

	isTranspiledInstanceOf(name: string): boolean {
		return ['de.svws_nrw.asd.validate.schueler.ValidatorSses00SchuelerStammdatenErsteStaatsangehoerigkeit', 'de.svws_nrw.asd.validate.BasicValidator', 'de.svws_nrw.asd.validate.Validator'].includes(name);
	}

	public static readonly class = new Class<ValidatorSses00SchuelerStammdatenErsteStaatsangehoerigkeit>('de.svws_nrw.asd.validate.schueler.ValidatorSses00SchuelerStammdatenErsteStaatsangehoerigkeit');

}

export function cast_de_svws_nrw_asd_validate_schueler_ValidatorSses00SchuelerStammdatenErsteStaatsangehoerigkeit(obj: unknown): ValidatorSses00SchuelerStammdatenErsteStaatsangehoerigkeit {
	return obj as ValidatorSses00SchuelerStammdatenErsteStaatsangehoerigkeit;
}

File: svws-asd/src/main/java/de/svws_nrw/asd/validate/schueler/ValidatorSses00SchuelerStammdatenErsteStaatsangehoerigkeit.java
package de.svws_nrw.asd.validate.schueler;

import java.util.function.Supplier;

import de.svws_nrw.asd.validate.Validator;
import de.svws_nrw.asd.validate.ValidatorKontext;
import de.svws_nrw.transpiler.annotations.AllowNull;
import jakarta.validation.constraints.NotNull;

/**
 * Dieser Validator führt eine Statistikprüfung auf die Staatsangehörigkeit bei den Stammdaten
 * eines Schuelers einer Schule aus.
 */
public final class ValidatorSses00SchuelerStammdatenErsteStaatsangehoerigkeit extends Validator {

	/** Die Staatsangehoerigkeit des Schuelers */
	private final @NotNull Supplier<@AllowNull Long> _idStaatsangehoerigkeit;

	/**
	 * Erstellt einen neuen Validator mit den übergebenen Daten und dem übergebenen Kontext
	 *
	 * @param idStaatsangehoerigkeit    StaatsangehörigkeitID
	 * @param kontext                   der Kontext des Validators
	 */
	public ValidatorSses00SchuelerStammdatenErsteStaatsangehoerigkeit(
			final @NotNull Supplier<@AllowNull Long> idStaatsangehoerigkeit,
			final @NotNull ValidatorKontext kontext) {
		super(kontext);
		this._idStaatsangehoerigkeit = idStaatsangehoerigkeit;
		_validatoren.add(new ValidatorSses01SchuelerStammdatenErsteStaatsangehoerigkeit(getNotNullSupplierLong(idStaatsangehoerigkeit), kontext));
	}

	@Override
	protected boolean pruefe() {
		final Long idStaatsangehoerigkeit = _idStaatsangehoerigkeit.get();

		if (idStaatsangehoerigkeit == null) {
			addFehler(0, "1. Staatsangehörigkeit des Schülers: Kein Wert vorhanden.");
			return false;
		}

		return true;
	}

}
